core/src/main/java/io/questdb/griffin/engine/orderby/LongSortedLightRecordCursorFactory.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.orderby;
+
+import io.questdb.cairo.AbstractRecordCursorFactory;
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.ColumnType;
+import io.questdb.cairo.ListColumnFilter;
+import io.questdb.cairo.sql.RecordCursor;
+import io.questdb.cairo.sql.RecordCursorFactory;
+import io.questdb.cairo.sql.RecordMetadata;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+
+/**
+ * Radix sort-based factory for ORDER BY over a single int/ipv4/long/timestamp/date column.
+ * Memory overhead is similar to {@link SortedRecordCursorFactory}:
+ * each row takes 32 bytes vs 36 bytes in the red-black tree-based factory.
+ */
+public class LongSortedLightRecordCursorFactory extends AbstractRecordCursorFactory {
+    private final RecordCursorFactory base;
+    private final LongSortedLightRecordCursor cursor;
+    private final ListColumnFilter sortColumnFilter;
+
+    public LongSortedLightRecordCursorFactory(
+            CairoConfiguration configuration,
+            RecordMetadata metadata,
+            RecordCursorFactory base,
+            ListColumnFilter sortColumnFilter
+    ) {
+        super(metadata);
+        this.base = base;
+        final int columnIndex = sortColumnFilter.getColumnIndexFactored(0);
+        this.cursor = new LongSortedLightRecordCursor(
+                configuration,
+                columnIndex,
+                metadata.getColumnType(columnIndex),
+                sortColumnFilter.getColumnIndex(0) > 0
+        );
+        this.sortColumnFilter = sortColumnFilter;
+    }
+
+    public static void addSortKeys(PlanSink sink, ListColumnFilter filter) {
+        sink.attr("keys").val('[');
```

Comment:
there seem to be code duplication here

Reply:
Not sure I follow. What kind of code duplication are you referring to?

Reply:
I've noticed the duplication now. Removed it in 6f911157d7d06a5dd30bf38f543ba86b1f5c15ba

Interestingly, my IDEA CE didn't highlight this method.